app/models/post.rb

```diff
@@ -1,4 +1,6 @@
 class Post < ApplicationRecord
+  validates :comment_counter, :likes_counter, numericality: { greater_than_or_equal_to: 0 }
+  validates :title, presence: true, length: { maximum: 250 }
   belongs_to :author, class_name: 'User'
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
```

Comment:
Please solve the `N+1` problem in the `index` action of the `Posts` controller when fetching all posts and their comments for a user. You may learn about the N+1 problem [here](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh#:~:text=What%20is%20the%20N%2B1,the%20result%20of%20that%20data.).